File: leetcode/fun with ararys 101/findConsecutiveones.c
#include <stdio.h>
void findMaxConsecutiveOnes(int* nums, int numsSize) {
    int max_ones = 0;
    int count = 0;
    // THIS IS ANOTHER APPROACH AND IT IS ALSO WORK
    // for(int i = 0; i < numsSize; i++ ){
    //     if(nums[i]) {
    //         max_ones = ++count > max_ones ? count : max_ones;
    //     } else {
    //         count = 0;
    //     }
    // }
    for(int i = 0; i < numsSize; i++ ){
        if(nums[i] == 1) {
            count++;
        } else {
            if(count > max_ones) {
                max_ones = count;
            }
            count = 0;
        }
    }
    if(count > max_ones){
        max_ones = count;
    }
    printf("largest consecutive oness: %d ", max_ones);
}

void main(){
    int nums[] = {1,1,0,1,1,1};
    findMaxConsecutiveOnes(nums, 6);
}

File: leetcode/fun with ararys 101/findConsecutiveones.py
from typing import List
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_ones = 0
        count = 0
        for i in nums:
            if i == 1:
                count = count + 1
            else:
                if count > max_ones:
                    max_ones = count
                count = 0
        if count > max_ones:
            max_ones = count
        return max_ones
                    
        